cms/djangoapps/contentstore/utils.py

```diff
@@ -296,7 +296,7 @@ def get_group_display_name(user_partitions, xblock_display_name):
     """
     for user_partition in user_partitions:
         for group in user_partition['groups']:
-            if str(group['id']) in xblock_display_name:
+            if _(u'Group ID {group_id}').format(group_id=group['id']) == xblock_display_name:
```

Comment:
I think you should at least only iterate over content experiment user partitions:

from xmodule.split_test_module import get_split_user_partitions